### ISSUE TITLE
cvxpy + ecos2.0.8 no longer compatible with numpy<1.20

### ISSUE THREAD

**Issue by roberthuisman** [cvxpy/cvxpy#1596]
Hi all,

I'm experiencing the same bug as I reported a year ago: https://github.com/cvxpy/cvxpy/issues/1229. This time around it is actually the ecos package (v2.0.8) that is built on numpy 1.20 and therefore is incompatible with environments that installed numpy<1.20. This bug makes cvxpy unusable.

ecos also needs a pyproject.toml, just like cvxpy has.
I submitted a PR here: https://github.com/embotech/ecos-python/pull/37

But I need someone to kick-off the testing pipeline (@SteveDiamond ? 😄  )

ps: OSQP is also missing this pyproject.toml. Would be good to have it there as well!

Thanks! Rob

**Comment by SteveDiamond** [comment]
@roberthuisman could you also check diffcp and scs to see if they are compatible with numpy<1.20?

**Comment by SteveDiamond** [comment]
@roberthuisman could you try ecos again?

**Comment by roberthuisman** [comment]
@SteveDiamond 

For ecos 2.0.9 i ran two test cases
1) Databricks runtime 7.3 LTS (https://docs.databricks.com/release-notes/runtime/7.3ml.html)
TLDR; python 3.7, numpy 1.18.1, pandas 1.0.1, cvxpy 1.1.18, cvxopt 1.2.7
2) Databricks runtime 9.1 LTS
python 3.8, numpy 1.19.2, pandas 1.2.4, cvxpy1.1.18, cvxopt 1.2.7

In both test cases cvxpy cannot find the ECOS installation? I now get the error message; 

def _find_candidate_solvers(self,
                                solver=None,
                                gp: bool = False):
        """
        Find candidate solvers for the current problem. If solver
        is not None, it checks if the specified solver is compatible
        with the problem passed.

        ---------
        solver : Union[string, Solver, None]
            The name of the solver with which to solve the problem or an
            instance of a custom solver. If no solver is supplied
            (i.e., if solver is None), then the targeted solver may be any
            of those that are installed. If the problem is variable-free,
            then this parameter is ignored.
        gp : bool
            If True, the problem is parsed as a Disciplined Geometric Program
            instead of as a Disciplined Convex Program.
    
        Returns
        -------
        dict
            A dictionary of compatible solvers divided in `qp_solvers`
            and `conic_solvers`.
    
        Raises
        ------
        cvxpy.error.SolverError
            Raised if the problem is not DCP and `gp` is False.
        cvxpy.error.DGPError
            Raised if the problem is not DGP and `gp` is True.
        """
        candidates = {'qp_solvers': [],
                      'conic_solvers': []}
        if isinstance(solver, Solver):
            return self._add_custom_solver_candidates(solver)
    
        if solver is not None:
            if solver not in slv_def.INSTALLED_SOLVERS:
>               raise error.SolverError("The solver %s is not installed." % solver)
E               cvxpy.error.SolverError: The solver ECOS is not installed.



scs is fine; see https://github.com/bodono/scs-python/pull/39/files
osqp i just requested a merge https://github.com/osqp/osqp-python/pull/67/files
diffcp i just request a merge https://github.com/cvxgrp/diffcp/pull/52

**Comment by SteveDiamond** [comment]
@roberthuisman can you try again? ecos should be fixed. There was a bug going back to ecos 2.0.8 actually...

**Comment by roberthuisman** [comment]
@SteveDiamond my issues are resolved with ecos2.0.10. Many thanks for the swift fixes :)
